### PR TITLE
Add support for reading the MDNS

### PR DESCRIPTION
**Pull request by lordyavin**
For the integration into Home Assistant, entities need to have a unique ID. IP address host or device name are not accepted. The "mdns" field returned by `/esp/get_identification` seams to be the MAC address prefixed with "ihs". MAC addresses are accepted as uniqe IDs.

```json
{
    "ip": "192.168.1.194",
    "name": "MyStove",
    "mdns": "ihs_5aea42b26597"
}
```

```
firmware_version: 3.23.0
algorithm: DS.DSB2.221129
wifi: 12.6.0
remote: 3.6.0
```

### CODE REVIEW

**Comment by mvn23** [comment]
Thanks for the PR. It has been a while since I looked at this repo and I am unable to test any of this at the moment (maybe I will regain access to a stove in a few months), but I am happy to review, merge and release a new version as we're still in alpha anyway :man_shrugging: 
However, I do like to keep things orderly, even at this stage. Could you remove the devcontainer stuff from this PR and split the version update into a separate PR? Thanks again!

**Comment by lordyavin** [comment]
Done

File: pystove/pystove.py [hunk 1]
```diff
@@ -382,11 +384,13 @@ async def get_name_and_ip():
             """Get stove name and IP."""
             stove_id = await self._get_json('http://' + self.stove_host
                                             + STOVE_ID_URL)
-            if None in [stove_id.get(DATA_NAME), stove_id.get(DATA_IP)]:
-                _LOGGER.warning("Unable to read stove name and/or IP.")
+            if None in [stove_id.get(DATA_NAME), stove_id.get(DATA_IP), stove_id[DATA_MDNS]]:
```
Comment (mvn23) [inline pystove/pystove.py]: `DATA_MDNS` is not guaranteed to be in the response. Use `stove_id.get(DATA_MDNS)` instead.

Also, is there a minimum stove firmware version where this value is present? If so, we should handle the case where mdns is not available but the other info is.

File: pystove/version.py [hunk 2]
```diff
@@ -14,4 +14,4 @@
 # along with pystove.  If not, see <http://www.gnu.org/licenses/>.
 #
 # Copyright 2019 Milan van Nugteren
-__version__ = '0.2a1'
+__version__ = '0.3a1'
```
Comment (mvn23) [inline pystove/version.py]: Please split out the version bump to a separate PR.

**Review by mvn23 — CHANGES_REQUESTED** [review]
Thanks for the update, please see the review notes I left.

File: pystove/version.py (hunk 2 above)
Comment (lordyavin) [reply pystove/version.py]: > Please split out the version bump to a separate PR.

Why?

Comment (mvn23) [reply pystove/version.py]: Each PR should be as small as possible and fulfill a single purpose. See [PR best practices](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/best-practices-for-pull-requests#write-small-prs)

Comment (lordyavin) [reply pystove/version.py]: Actually the increment of the minor version reflects the "feature" addition and is directly related to the changes of this PR. So IMHO it is totally reasonable to increase it along  with adding the feature. If you insist to remove it I will do so.

Comment (mvn23) [reply pystove/version.py]: By that logic, adding 3 new features in quick succession between 2 releases should increase the minor version by 3 as well...
As it is now this PR adds a feature AND increases the version number, fulfilling more than 1 purpose and making it larger than necessary (although still pretty small in general ;) ).

Comment (lordyavin) [reply pystove/version.py]: Which are the 2 other features? The devcontainer PR should not increase the library version as it is just about the development environment. It changes nothing in the library source code.

Anyway. I'm going to remove it. At the end you are in charge to do the release. I'm crossing my fingers that I don't have to wait for long 😉

File: pystove/pystove.py (hunk 1 above)
Comment (lordyavin) [reply pystove/pystove.py]: > is there a minimum stove firmware version where this value is present

I don't have such insights. My stove's airbox reports the following sw versions:
firmware_version: 3.23.0
algorithm: DS.DSB2.221129
wifi: 12.6.0
remote: 3.6.0

**Review by mvn23 — APPROVED** [review]
Awesome, thanks!

File: pystove/pystove.py (hunk 1 above)
Comment (mvn23) [reply pystove/pystove.py]: To follow-up on this, as I now have access to a stove again, I see the following values:

firmware 3.23.0
algorithm HW.3600.221202
wifi 12.5.0
remote 1.12.0

This version does not provide mdns information in `/esp/get_stove_data`, so using it as an identifier in Home Assistant is not fool-proof...

Comment (lordyavin) [reply pystove/pystove.py]: I assume the algorithm has nothing to do with it and the remote too. Firmware is equal to mine so looks like the Wi-Fi is the sw part that implements the REST interface. 

Question is if there is any other unique identification. 

BTW where did you get all the infos about the API? Sniffing?
